fix: improved indentation for code samples

scottrippey opened the pull request:
<!--

Have you read Formidable's Code of Conduct? By filing an Issue, you are expected to comply with it, including treating everyone with respect: https://github.com/FormidableLabs/spectacle/blob/main/CONTRIBUTING.md#contributor-covenant-code-of-conduct

-->

### Description

Fixed indentation in the code blocks on the one-page example.


Fixes # (issue)

#### Type of Change

Please delete options that are not relevant (including this descriptive text).

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

### How Has This Been Tested?

### Before:

![image](https://user-images.githubusercontent.com/430608/140995335-53dff6e3-6c72-4949-9e8a-31f3ef682aba.png)

### After:

![image](https://user-images.githubusercontent.com/430608/140995397-333b45d5-8b76-4be8-ae5b-a98a1c184306.png)

ryan-roemer commented:
Was this manually edited or with `yarn build-one-page` which is our generation tool?

scottrippey commented:
Manually edited!  Didn't know it was generated.  So yeah, my changes will probably get overwritten with this approach.
I'll look into fixing the `build-one-page` script instead.

ryan-roemer commented:
Ah, I remembered -- we already have a bug for it: https://github.com/FormidableLabs/spectacle/issues/913

scottrippey commented:
Update: fixed the `build-one-page` script to retain proper indentation.

The committed `one-page.html` was generated by the script.

I also did a visual inspection of the "Markdown" slides, since the indentation also changed there, and they're identical before / after:
![image](https://user-images.githubusercontent.com/430608/141005387-a729349c-0b29-4941-a1ea-4a14e900b610.png)

scottrippey commented on scripts/one-page.js:
```diff
@@ -53,7 +53,16 @@ const getSrcContent = async src => {
+        .replace(/`}\n/gm, '`}######</pre>\n');
+
+      // Make the HTML pretty:
+      htm = pretty(htm).replace(/<pre>######|######<\/pre>/g, '');
```
FWIW, I added the `######` to ensure we don't accidentally delete a real `<pre>` tag.

ryan-roemer replied:
Great Idea! 

It's out there, but I could see `######` potentially happening in a deck. Can we maybe switch to a much more obnoxious token like `SPECTACLE_ONE_PAGE_TEMP_PRE_TOKEN_START` and `SPECTACLE_ONE_PAGE_TEMP_PRE_TOKEN_STOP` or something?

scottrippey replied:
Absolutely, good idea.

scottrippey commented:
@ryan-roemer I'll wait for an official stamp of approval before merging.  Post-merge, what needs to be done?

ryan-roemer reviewed (approved):
Looks great! Thanks so much for this fix!!!

One minor refactor, after which I don't need re-review.

ryan-roemer commented:
@scottrippey Post-merge, we can either do nothing, or do a patch release. This would mean that the `unpkg` version at: https://unpkg.com/spectacle@8.4.0/examples/one-page.html looks better, but it's not a huge deal.

@carlos-kelly what do you think?

carlos-kelly commented:
@scottrippey @ryan-roemer I think a patch version is great. I can cut one tomorrow.
